packages/language-server/src/completion/completions.ts

```diff
@@ -865,40 +857,86 @@ function getFieldType(line: string): string | undefined {
   return undefined
 }
 
+// function definingReferentialAction(
+//   wordsBeforePosition: Array<string>,
+// ): boolean {
+//   const lastWord = wordsBeforePosition[wordsBeforePosition.length - 2]
+//   return (
+//     lastWord != undefined &&
+//     (lastWord.includes('onDelete') || lastWord.includes('onUpdate'))
+//   )
+// }
+
+// @relation
 function getSuggestionsForRelationDirective(
   wordsBeforePosition: string[],
   currentLineUntrimmed: string,
   lines: string[],
-  document: TextDocument,
+  document: TextDocument, // eslint-disable-line @typescript-eslint/no-unused-vars
   block: Block,
   position: Position,
-  binPath: string,
+  binPath: string, // eslint-disable-line @typescript-eslint/no-unused-vars
 ): CompletionList | undefined {
   // create deep copy
   const suggestions: CompletionItem[] = klona(relationArguments)
-  const wordBeforePosition = wordsBeforePosition[wordsBeforePosition.length - 1]
+  const firstWordBeforePosition =
+    wordsBeforePosition[wordsBeforePosition.length - 1]
+  const secondWordBeforePosition =
+    wordsBeforePosition[wordsBeforePosition.length - 2]
+  const wordBeforePosition =
+    firstWordBeforePosition === ''
+      ? secondWordBeforePosition
+      : firstWordBeforePosition
```

Comment:
Why is this necessary? Why can `firstWordBeforePosition` be an empty string?

Reply:
because `onDelete: |` if the cursor is here there first word is empty, which is what happens most of the time.

Reply:
Is that our logic that splits for the words or something built in? Feels super weird.

Reply:
Yes it's custom code 
```
const wordsBeforePosition: string[] = currentLineTillPosition.split(/\s+/)
```